key_pair.go

```diff
@@ -0,0 +1,315 @@
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return &k, resp, err
+}
+
+// UpdateByRepoId updates the key pair variable based on the given option
+//
+// Travis CI API docs: https://developer.travis-ci.com/resource/key_pair#update
+func (ks *KeyPairService) UpdateByRepoId(ctx context.Context, repoId uint, keyPair *KeyPairBody) (*KeyPair, *http.Response, error) {
+	u, err := urlWithOptions(fmt.Sprintf("/repo/%d/key_pair", repoId), nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := ks.client.NewRequest(http.MethodPatch, u, keyPair, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var k KeyPair
+	resp, err := ks.client.Do(ctx, req, &k)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return &k, resp, err
+}
+
+// UpdateByRepoSlug updates the key pair variable based on the given option
+//
+// Travis CI API docs: https://developer.travis-ci.com/resource/key_pair#update
+func (ks *KeyPairService) UpdateByRepoSlug(ctx context.Context, repoSlug string, keyPair *KeyPairBody) (*KeyPair, *http.Response, error) {
+	u, err := urlWithOptions(fmt.Sprintf("/repo/%s/key_pair", url.QueryEscape(repoSlug)), nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := ks.client.NewRequest(http.MethodPatch, u, keyPair, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var k KeyPair
+	resp, err := ks.client.Do(ctx, req, &k)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return &k, resp, err
+}
+
+// DeleteByRepoId deletes the key pair based on the given repository id and
+//
+// Travis CI API docs: https://developer.travis-ci.com/resource/key_pair#delete
+func (ks *KeyPairService) DeleteByRepoId(ctx context.Context, repoId uint) (*http.Response, error) {
+	u, err := urlWithOptions(fmt.Sprintf("/repo/%d/key_pair", repoId), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := ks.client.NewRequest(http.MethodDelete, u, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := ks.client.Do(ctx, req, nil)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, err
+}
+
+// DeleteByRepoSlug deletes the key pair based on the given repository slug
+//
+// Travis CI API docs: https://developer.travis-ci.com/resource/key_pair#delete
+func (ks *KeyPairService) DeleteByRepoSlug(ctx context.Context, repoSlug string) (*http.Response, error) {
+	u, err := urlWithOptions(fmt.Sprintf("/repo/%s/key_pair", url.QueryEscape(repoSlug)), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := ks.client.NewRequest(http.MethodDelete, u, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := ks.client.Do(ctx, req, nil)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, err
+}
+
+// FindGeneratedByRepoId fetches the default key pair based on the given repository id
+//
+// Travis CI API docs: https://developer.travis-ci.com/resource/key_pair_generated#find
+func (ks *KeyPairService) FindGeneratedByRepoId(ctx context.Context, repoId uint) (*KeyPair, *http.Response, error) {
```

Comment:
Would you create a new service like `GeneratedKeyPairService` and move `FindGenerated~` and `CreateGenerated~` there? I would like to keep the library's api coherent, which method names do not contain information about resource they interact with.

Reply:
Make sense 😃  I fixed it at cba3866

Reply:
Thanks for the fix! Looks great! 👍